### ISSUE TITLE
Bug: DropwizardMeterRegistry does not remove Gauges created by LongTaskTimer

### ISSUE THREAD

**Issue by hendryluk** [micrometer-metrics/micrometer#2924]
## Steps to reproduce:
1. Create a LongTaskTime onto a DropwizardMeterRegistry (e.g. JMX)
   ```java
   MeterRegistry registry = newJmxMeterRegistry(JmxConfig.DEFAULT, Clock.SYSTEM);
   LongTaskTimer timer = registry.more().longTaskTimer("foo"); 
   timer.record(() => {});
   ```
   In JMX, this creates 3 gauge mbeans: `foo.statistic.active`, `foo.statistic.max`, and `foo.statistic.duration`.
2. Remove the timer
    ```java
    registry.remove(timer);
    timer.close();
    ```

### Expected:
It should remove all those 3 gauges from the underlying registry (i.e. JMX in this case).

### Actual:
All the 3 gauges associated with the LongTimerMetric remain in JMX

## Cause:
[DropwizardMeterRegistry#newLongTaskTimer](https://github.com/micrometer-metrics/micrometer/blob/bfdb43e3044861bbfcad0563b11f673c30af3a0f/micrometer-core/src/main/java/io/micrometer/core/instrument/dropwizard/DropwizardMeterRegistry.java#L120) creates 3 gauges (Active, Duration, and Max). However it never gets removed during [DropwizardMeterRegistry#onMeterRemoved](https://github.com/micrometer-metrics/micrometer/blob/bfdb43e3044861bbfcad0563b11f673c30af3a0f/micrometer-core/src/main/java/io/micrometer/core/instrument/dropwizard/DropwizardMeterRegistry.java#L71)

## Fix
In [DropwizardMeterRegistry#onMeterRemoved](https://github.com/micrometer-metrics/micrometer/blob/bfdb43e3044861bbfcad0563b11f673c30af3a0f/micrometer-core/src/main/java/io/micrometer/core/instrument/dropwizard/DropwizardMeterRegistry.java#L71), add the following:
```
if (meter instanceof LongTaskTimer) {
   for (Statistic statistic: Statistic.values()) {
      registry.remove(hierarchicalName(meter.getId().withTag(statistic)));
   }
}
```

**Comment by shakuzen** [comment]
Thanks for the detailed report. It sounds like this should also be a problem for removing custom `Meter` implementations and `Timer`/`DistributionSummary` with histogram gauges configured.

Would you be interested and willing to submit a pull request for the fix to the `1.7.x` branch?

**Comment by marcingrzejszczak** [comment]
Hey @hendryluk maybe you would interested in filing the PR with the fix for `1.9.x` branch?
